src/megatron/energon/task_encoder/base.py

```diff
@@ -229,19 +292,42 @@ def build_batch(
                 worker_config=worker_config,
             )
         elif batch_size is not None:
-            dataset = BatchDataset(
-                dataset,
-                batch_size=batch_size,
-                batcher=self.batch,
-                drop_last=batch_drop_last,
-                worker_config=worker_config,
-            )
+            if getattr(self.slice_batch, "__func__", None) is not TaskEncoder.slice_batch:
+                assert not batch_drop_last, "batch_drop_last is not supported if slicer is set"
+                dataset = SliceBatchDataset(
```

Comment:
TODO: new: keep batch dataset, just insert the SlicePackDataset if enabled (buffer>0 and Method overridden).